[Doc] Add documentation for getting started and dev with learning engine

goldenleaves opened the pull request:
<!--
Thanks for your contribution! please review https://github.com/alibaba/GraphScope/blob/main/CONTRIBUTING.md before opening an issue.
-->

## What do these changes do?

Add the documentation for GLE:
(1) getting started
(2) dev and tes

## Related issue number

<!-- Are there any issues opened that will be resolved by merging this change? -->

Related to #2522

siyuan0322 reviewed (approved):
I changed the `fixes #xxx` to `related to #xxx` since `fixes` would close the whole issue. You may check the corresponding checkbox of that issue.

goldenleaves commented:
> I changed the `fixes #xxx` to `related to #xxx` since `fixes` would close the whole issue. You may check the corresponding checkbox of that issue.

checked

> I changed the `fixes #xxx` to `related to #xxx` since `fixes` would close the whole issue. You may check the corresponding checkbox of that issue.

checked
